src/components/App/App.js

```diff
@@ -26,12 +31,37 @@ const PAGES_TO_SHOW_TIMEOUT = [
 ];
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    initFirebase();
```

Comment:
`index.js` might be a good place for this since that's where we do other project setup (e.g. `ReactDOM.render`)

Reply:
With components mounting and unmounting all the time, I noticed a lot of issues with the firebase initialisation. So I wrote the initFirebase function which only inits it if it isn't already and I've been calling that before all firebase actions.